src/lib/swapper/swappers/LifiSwapper/utils/transformLifiFeeData/transformLifiFeeData.ts

```diff
@@ -1,77 +1,57 @@
 import { bn } from 'lib/bignumber/bignumber'
-import type { ProtocolFee, QuoteFeeData } from 'lib/swapper/api'
+import type { ProtocolFee } from 'lib/swapper/api'
 import { SwapError, SwapErrorType } from 'lib/swapper/api'
-import { getFeeAssets } from 'lib/swapper/swappers/LifiSwapper/utils/getFeeAssets/getFeeAssets'
-import { processGasCosts } from 'lib/swapper/swappers/LifiSwapper/utils/processGasCosts/processGasCosts'
 import { selectAssets } from 'state/slices/selectors'
 import { store } from 'state/store'
 
 import { lifiChainIdToChainId } from '../lifiChainIdtoChainId/lifiChainIdToChainId'
 import { lifiTokenToAssetId } from '../lifiTokenToAssetId/lifiTokenToAssetId'
 
-export const transformLifiFeeData = ({
+export const transformLifiStepFeeData = ({
   chainId,
-  selectedRoute,
+  lifiStep,
 }: {
-  buyAsset: Asset
   chainId: ChainId
-  selectedRoute: Route
-}): QuoteFeeData<EvmChainId> => {
+  lifiStep: Step
+}): Record<AssetId, ProtocolFee> => {
   if (!isEvmChainId(chainId)) {
     throw new SwapError("[transformLifiFeeData] - chainId isn't an EVM ChainId", {
       code: SwapErrorType.UNSUPPORTED_CHAIN,
       details: { chainId },
     })
   }
 
-  const allRouteGasCosts = selectedRoute.steps.flatMap(step => step.estimate.gasCosts ?? [])
-  const allRouteFeeCosts = selectedRoute.steps
-    .flatMap(step => step.estimate.feeCosts ?? [])
-    .filter(feeCost => !(feeCost as { included?: boolean }).included) // filter out included fees
-
-  const feeAsset = getFeeAssets(chainId)
-
-  const { networkFeeCryptoBaseUnit, otherGasCosts } = processGasCosts({
-    feeAsset,
-    allRouteGasCosts,
-  })
+  // filter out included fees
+  const allPayableFeeCosts = (lifiStep.estimate.feeCosts ?? []).filter(
+    feeCost => !(feeCost as { included?: boolean }).included,
+  )
 
   const assets = selectAssets(store.getState())
 
-  // TEMP: jumble in gas costs not denominated in the sell chain fee asset with protocol fees
-  // until our UI can convey the concept of gas fees denominated in multiple tokens
-  const protocolFees = [...allRouteFeeCosts, ...otherGasCosts].reduce<Record<AssetId, ProtocolFee>>(
-    (acc, feeCost) => {
-      const { amount, token } = feeCost
-      const assetId = lifiTokenToAssetId(token)
-      const asset = assets[assetId]
-      if (acc[assetId] === undefined) {
-        acc[assetId] = {
-          amountCryptoBaseUnit: amount,
-          asset: {
-            chainId: asset?.chainId ?? lifiChainIdToChainId(token.chainId),
-            precision: asset?.precision ?? token.decimals,
-            symbol: asset?.symbol ?? token.symbol,
-          },
-          requiresBalance: true,
-        }
-      } else {
-        acc[assetId].amountCryptoBaseUnit = bn(acc[assetId].amountCryptoBaseUnit)
-          .plus(amount)
-          .toString()
+  const protocolFees = allPayableFeeCosts.reduce<Record<AssetId, ProtocolFee>>((acc, feeCost) => {
+    const { amount, token } = feeCost
```

Comment:
We might want to destructure this as below for clarity:

```suggestion
    const { amount: amountCryptoBaseUnit, token } = feeCost
```